src/FritzBox/Converter.php

```diff
@@ -132,79 +149,72 @@ private function addEmail(array $addresses)
      * Return an array of prequalified phone numbers. This is neccesseary to
      * handle the maximum of nine phone numbers per FRITZ!Box phonebook contacts
      *
-     * @param stdClass $card
+     * @param mixed $card
      * @return array
      */
-    private function getPhoneNumbers(stdClass $card): array
+    private function getPhoneNumbers($card): array
     {
-        if (!isset($card->phone)) {
+        if (!isset($card->TEL)) {
             return [];
         }
 
-        $res = [];
+        $phoneNumbers = [];
 
         $replaceCharacters = $this->config['phoneReplaceCharacters'] ?? [];
         $phoneTypes = $this->config['phoneTypes'] ?? [];
-
-        foreach ($card->phone as $numberType => $numbers) {
-            foreach ($numbers as $number) {
-                // format number
-                if (count($replaceCharacters)) {
-                    $number = str_replace("\xc2\xa0", "\x20", $number);   // delete the wrong ampersand conversion
-                    $number = strtr($number, $replaceCharacters);
-                    $number = trim(preg_replace('/\s+/', ' ', $number));
+        foreach ($card->TEL as $key => $number) {
+            // format number
+            if (count($replaceCharacters)) {
+                $number = str_replace("\xc2\xa0", "\x20", $number);
+                $number = strtr($number, $replaceCharacters);
+                $number = trim(preg_replace('/\s+/', ' ', $number));
+            }
+            // get type
+            $type = 'other';
+            $telTypes = isset($card->TEL[$key]->parameters['TYPE']) ? strtoupper($card->TEL[$key]->parameters['TYPE']) : '';
```

Comment:
You can use the ?: operator to simplify this.

Reply:
cool - the [Elvis operator](https://stackoverflow.com/questions/1993409/operator-the-elvis-operator-in-php) :)
But I´m checking against isset() and need to get strtoupper() - so I´m a little bit puzzled how this fits into the shortend syntax

Reply:
Elvis? Nice. You can use Elvis inside and wrap then:

    $types = strtoupper(... ?: „“)

A little less efficient but more readable.

Reply:
> Good idea - wrong method
(Douglas Adams - Last chance to see)

First time I consent your tests: change to
`$telTypes = strtoupper(isset($card->TEL[$key]->parameters['TYPE']) ?: '');`
results with 3 errors

Reply:
So schwer ist es doch nicht wenn Du Dir anschaust was ?: tut. Das isset ist doch schon eingebaut!

    $telTypes = strtoupper($card->TEL[$key]->parameters['TYPE'] ?: '');

Reply:
und warum geht dann das script auf die Bretter (na ja NOTICE:), wenn ich Kontakte ohne Rufnummer durch den Konverter jage?

Reply:
Schwer zu sagen ohne den Code mit Testcase zu sehen. Dafür sind die da.